code/modules/surgery/organs/internal/cyberimp/augments_chest.dm

```diff
@@ -119,16 +112,21 @@
 
 
 /obj/item/organ/internal/cyberimp/chest/reviver/proc/revive_dead()
-	owner.grab_ghost()
-
-	owner.visible_message(span_warning("[owner]'s body convulses a bit."))
-	playsound(owner, SFX_BODYFALL, 50, TRUE)
-	playsound(owner, 'sound/machines/defib_zap.ogg', 75, TRUE, -1)
-	owner.revive()
-	owner.emote("gasp")
-	owner.set_jitter_if_lower(200 SECONDS)
-	SEND_SIGNAL(owner, COMSIG_LIVING_MINOR_SHOCK)
-	log_game("[owner] been revived by [src]")
+	if(COOLDOWN_FINISHED(src, defib_cooldown) && owner.stat == DEAD && owner.can_defib() == DEFIB_POSSIBLE)
```

Comment:
Do an early return please.

Reply:
Alright.

Reply:
I believe the change I made is what you wanted, if not let me know..